classes/01class/exercises/c01-aws01/nimmi89/README.md

```diff
@@ -0,0 +1,440 @@
+                    "VpcId": "vpc-12497675",
+                    "InterfaceType": "interface"
+                }
+            ],
+            "RootDeviceName": "/dev/xvda",
+            "RootDeviceType": "ebs",
+            "SecurityGroups": [
+                {
+                    "GroupName": "C01-AWS01SSHAccess",
+                    "GroupId": "sg-0619b1745d16f5cb6"
+                }
+            ],
+            "SourceDestCheck": true,
+            "StateReason": {
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "438549961569",
+    "ReservationId": "r-0f504e968c8d0a14f"
+}
+```
+
+- Commands to connect to the first EC2 instance:
+```
+>>>Command to describe your instances using the instance id returned in the above command
+aws ec2 describe-instances \
+--instance-id i-05d7620963b672ca0
+
+>The json output shows instance in running state along with other parameters.
+ "PublicIpAddress": "52.64.28.243",
+                    "State": {
+                        "Code": 16,
+                        "Name": "running"
+
+>>>Command to SSH into the instance with the public ip returned and the key pair you created.
+ssh -i C01-AWS01KeyPair.pem ec2-user@52.64.28.243
+
+Note: If it returns an error ec2-user@52.64.28.243: Permission denied (publickey,gssapi-keyex,gssapi-with-mic)., you need to provide permmissions to read your SSH key
+chmod 400 C01-AWS01KeyPair.pem
+
+Run SSH Command again to get into EC2 Instance successfully!!!!!!!!!!!
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-32-188 ~]$ uname -a
+Linux ip-172-31-32-188.ap-southeast-2.compute.internal 4.14.177-139.254.amzn2.x86_64 #1 SMP Thu May 7 18:48:23 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
+```
+
+- Commands to create the second EC2 instance and any additional resource required:
+```
+>>>Command to launch a private instance[no-associate-public-ip-address] into your subnet, using the security group and key pair you've created.The ami id should be according to your region. 
+Can use the same security group,key-pair or create new one .
+aws ec2 run-instances \
+--image-id ami-088ff0e3bde7b3fdf --count 1 \
+--instance-type t2.micro --key-name C01-AWS01KeyPair \
+--security-group-ids sg-0619b1745d16f5cb6
```

Comment:
Same as the previous one.. any idea on how you can improve the security in this security group? 
remember you can make a dedicated security group for each instance..

Reply:
I have made the changes to allow this security group to allow SSH only from the public instance's security group!